File: app14_LUA_UNITY/backend/app/models/mitigation.lua
local Model = require("lapis.db.model").Model

local Mitigation = Model:extend("mitigations", {
  primary_key = "slug",

  relations = {
    { "threat", belongs_to = "Threat", key = "threat_code" },
    { "card", belongs_to = "Card", key = "card_id" },
    { "code_samples", has_many = "CodeSample", key = "mitigation_slug" }
  }
})

return Mitigation

File: app13_ruby_FastApi/backend/app/models/mitigation.rb
# frozen_string_literal: true

class Mitigation < Sequel::Model(:mitigations)
  unrestrict_primary_key

  many_to_one :threat, key: :threat_code
  many_to_one :card, key: :card_id
  one_to_many :code_samples, key: :mitigation_slug
end
